### PR TITLE
Replace partitions actuator with Feign based client

### PR DESCRIPTION
**Pull request by npepinpe**
## Description

This PR replaces the old `PartitionsActuatorClient` with a simple interface + Feign based client. The idea is to introduce an easy way to build actuator/management clients focused on ease of maintenance and ergonomics. Adding a new method means simply adding it in the interface and nothing else. No more either - I think this is a good pattern for production code, but for tests I'd rather fail fast with a descriptive exception than deal with `Either`.

I would propose we follow this pattern in the future as well for other actuators.

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: qa/util/src/main/java/io/camunda/zeebe/qa/util/actuator/PartitionsActuator.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    return Feign.builder()
+        .encoder(new JacksonEncoder())
+        .decoder(new JacksonDecoder())
+        .retryer(Retryer.NEVER_RETRY)
+        .target(target);
+  }
+
+  @RequestLine("GET /")
+  @Headers("Accept: application/json")
+  Map<Integer, PartitionStatus> query();
```
Comment (npepinpe) [inline qa/util/src/main/java/io/camunda/zeebe/qa/util/actuator/PartitionsActuator.java:71]: :thought_balloon: I did think about going with `get` as a name, but it seemed too generic. Let me know what you think.

**Review by lenaschoenburg — APPROVED** [review]
Nice, looks good to me! :tada: 

:+1: Good call on just not using `Either` in tests, they are easier to read now
:+1: Thanks for documenting `PartitionActuator` so extensively

**Comment by npepinpe** [comment]
bors merge

**Comment by npepinpe** [comment]
bors r+

The failing test is a known flaky test.

**Comment by npepinpe** [comment]
Alright #10097 should be merged before we backport this, otherwise it's too much of a pain.

**Pull request merged**